src/plots/cartesian/axes.js

```diff
@@ -2296,16 +2296,35 @@ axes.draw = function(gd, arg, opts) {
             var id = d[0];
             var plotinfo = fullLayout._plots[id];
             if(plotinfo) {
+                var xa = plotinfo.xaxis;
                 var ya = plotinfo.yaxis;
-                if(ya.tickmode === 'sync' && ya.overlaying === arg[0]) {
-                    arg.push(ya._id);
-                }
+                isSyncAxis(xa, arg[0]);
+                isSyncAxis(ya, arg[0]);
             }
         });
     }
 
+    function isSyncAxis(ax, idToValidate) {
```

Comment:
One may expect `isSyncAxis` to return true or false not modifying the `arg` array.
So let's rename this function e.g. to `addSyncAxis`.